Tests/UnitTests/Analytics/AnalyticsProviderTests.swift

```diff
@@ -59,10 +59,38 @@ class AnalyticsProviderTests: XCTestCase {
             configuration: analyticsConfiguration,
             eventDataSource: eventDataSource
         )
+        
+        let expectedCheckoutAttemptId = checkoutAttemptIdMockValue
+
+        let initialAnalyticsResponse = InitialAnalyticsResponse(checkoutAttemptId: expectedCheckoutAttemptId)
+        let checkoutAttemptIdResult: Result<Response, Error> = .success(initialAnalyticsResponse)
+        apiClient.mockedResults = [checkoutAttemptIdResult]
 
         // When
         sut.sendInitialAnalytics(with: .components(type: .affirm), additionalFields: nil)
-        XCTAssertEqual(sut.checkoutAttemptId, "do-not-track")
+        wait(until: sut, at: \.checkoutAttemptId, is: expectedCheckoutAttemptId)
+    }
+    
+    func testFetchCheckoutAttemptIdFailedResult() throws {
+        // Given
+        var analyticsConfiguration = AnalyticsConfiguration()
```

Comment:
This configuration and apiClient creation code is quite repetitive, let's extract it to an internal helper function?